examples/resources/group/main.tf

```diff
@@ -7,18 +7,70 @@ terraform {
 }
 
 provider "aap" {
-  host                 = "https://controller.xxx.xyz/"
-  username             = "xxxx"
-  password             = "xxxx"
+  host                 = "https://localhost:8043"
+  username             = "ansible"
+  password             = "test123!"
   insecure_skip_verify = true
 }
 
-resource "aap_group" "sample" {
+resource "aap_group" "sample_foo" {
   inventory_id = 1
-  name         = "tf_group"
+  name         = "tf_group_foo"
   variables    = jsonencode({ "ansible_network_os" : "ios" })
 }
 
-output "group" {
-  value = aap_group.sample
+locals {
+  values_variables = <<EOT
+exampleVariables:
+  - name: "bar"
+    namespace: "bar-namespace"
+    type: 0
+EOT
+}
+
+resource "aap_group" "sample_bar" {
+  inventory_id = 1
+  name         = "tf_group_bar"
+  variables    = jsonencode(yamldecode(local.values_variables))
+}
+
+resource "aap_group" "sample_baz" {
+  inventory_id = 1
+  name         = "tf_group_baz"
+  variables = jsonencode({
+    ansible_network_os = "ios"
+    # Add other variables as needed
+  })
+}
+
+resource "aap_group" "sample_abc" {
+  inventory_id = 1
+  name         = "tf_group_abc"
+  variables    = yamlencode({ "os" : "Linux", "automation" : "ansible" })
+}
+
+resource "aap_group" "sample_xyz" {
+  inventory_id = 1
+  name         = "tf_group_xyz"
+  variables    = "Not valid JSON"
+}
```

Comment:
Why include this one?

Reply:
This is a copy-paste issue. It should be "os: Linux\nautomation: ansible-devel". I'll fix that, Thanks.